bundle/config/validate/fast_validate.go

```diff
@@ -0,0 +1,51 @@
+package validate
+
+import (
+	"context"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+)
+
+// FastValidate runs a subset of fast validation checks. This is a subset of the full
+// suite of validation mutators that satisfy ANY ONE of the following criteria:
+//
+// 1. No file i/o or network requests are made in the mutator.
+// 2. The validation is blocking for bundle deployments.
+//
+// The full suite of validation mutators is available in the [Validate] mutator.
+type fastValidateReadonly struct{}
+
+func FastValidateReadonly() bundle.ReadOnlyMutator {
+	return &fastValidateReadonly{}
+}
+
+func (f *fastValidateReadonly) Name() string {
+	return "fast_validate(readonly)"
+}
+
+func (f *fastValidateReadonly) Apply(ctx context.Context, rb bundle.ReadOnlyBundle) diag.Diagnostics {
+	return bundle.ApplyReadOnly(ctx, rb, bundle.Parallel(
+		// Fast mutators with only in-memory checks
+		JobClusterKeyDefined(),
+		JobTaskClusterSpec(),
+		SingleNodeCluster(),
```

Comment:
Do all of these return errors?

Warnings currently only show up at the end of deploy, so they're not actionable (yet).

Reply:
Good point, `SingleNodeCluster()` returns a warning that is useful even with a deployment. The others, though indeed, are no longer useful.

I can follow up with a PR that serializes the warnings (and below like INFO) on a rolling basis in the `bundle.Seq` mutator. That should make the warnings useful before deployment.

Reply:
> The others, though indeed, are no longer useful.

How so?

Reply:
We need to revisit how we output these diags anyway, so no need to address here.

Reply:
> How so?

You'd end up with runtime errors on `JobClusterKeyDefined`. The validation warning has a runtime error equivalent:

```
.venv➜  bundle-playground git:(master) ✗ databricks bundle validate
Warning: job_cluster_key c1 is not defined
  at resources.jobs.foo.tasks[0].job_cluster_key
  in databricks.yml:17:28
```

```
Error: terraform apply: exit status 1

Error: cannot create job: Job cluster 'c1' is not defined in field 'job_clusters'.

  with databricks_job.foo,
  on bundle.tf.json line 35, in resource.databricks_job.foo:
  35:       }

```

`JobTaskClusterSpec` returns errors though so it's useful for returning an error early.

Reply:
We should probably promote this warning to an error though.